tools/build-tool/code/tooldebian.pas

```diff
@@ -0,0 +1,192 @@
+  strict private
+    { Total size of all binaries in the project, required for Debian metadata }
+    BinariesSize: Int64;
+    { Helper to find total size of all files inside the folder,
+      To satisfy FindFile syntax has to be of Object }
+    procedure FoundFile(const FileInfo: TFileInfo; var StopSearch: Boolean);
+  public
+    procedure Make(const OutputProjectPath: String; const PackageFileName: String;
+      const Cpu: TCpu; const Manifest: TCastleManifest);
+  end;
+
+implementation
+uses
+  SysUtils, Process, {$ifdef UNIX} BaseUnix, {$endif}
+  CastleUtils, CastleFilesUtils, CastleDownload, CastleImages,
+  ToolCommonUtils;
+
+procedure TPackageDebian.FoundFile(const FileInfo: TFileInfo; var StopSearch: Boolean);
+begin
+  BinariesSize += FileInfo.Size;
+end;
+
+procedure TPackageDebian.Make(const OutputProjectPath: String; const PackageFileName: String;
+      const Cpu: TCpu; const Manifest: TCastleManifest);
+
+  function CpuToArchitectureString(const Cpu: TCpu): String;
+  begin
+    case Cpu of
+      { Architectures supported by https://packages.debian.org/buster/bash }
+      x86_64: Result := 'amd64';
+      aarch64: Result := 'arm64';
+      arm: begin
+          Result := 'armel';
+          WriteLn('WARNING: Architecture ' + CpuToString(Cpu) + ' is ambiguous between "armel" and "armhf" in Debian.');
+        end;
+      i386: Result := 'i386';
+      mips: Result := 'mips';
+      mipsel: Result := 'mipsel';
+      powerpc64: begin
+          Result := 'ppc64el';
+          WriteLn('WARNING: Architecture ' + CpuToString(Cpu) + ' is ambiguous between "ppc64el" and "ppc64" in Debian.');
+        end;
+      // "mips64el"
+      // "s390x"
+      { Unofficial Debian ports https://packages.debian.org/sid/bash }
+      m68k: Result := 'm68k';
+      riscv64: Result := 'riscv64';
+      sparc64: Result := 'sparc64';
+      // "alpha"
+      // "sh4"
+      // "x32" = X32 is an ABI for amd64/x86_64 CPUs using 32-bit integers, longs and pointers
+      else {powerpc,sparc,avr,jvm,i8086,riscv32,armeb}
+      begin
+        Result := CpuToString(Cpu);
+        WriteLn('WARNING: Architecture ' + CpuToString(Cpu) + ' is not officially supported by Debian.');
+      end;
+    end;
+  end;
+
+var
+  PathToExecutableLocal, PathToExecutableUnix: String;
+  PathToIconFileLocal, PathToIconFileUnix: String;
+  ShareDir: String;
+  TextWriter: TTextWriter;
+  OutString: String;
+  PackageFolder: String;
+begin
+  PackageFolder := 'castle-engine-output' + PathDelim + PackageFileName;
+  if DirectoryExists(PackageFolder) then
+    RemoveNonEmptyDir(PackageFolder);
+
+  // Copy the binaries
+
+  PathToExecutableUnix := '/usr/' + Manifest.DebianInstallFolder + '/' + Manifest.Name;
+  PathToExecutableLocal := StringReplace(PathToExecutableUnix, '/', PathDelim, [rfReplaceAll]);
+  CopyDirectory(Path, PackageFolder + PathToExecutableLocal);
+  ShareDir := PackageFolder + PathDelim + 'usr' + PathDelim + 'share';
+  CreateDir(ShareDir);
```

Comment:
Check `CreateDir` result -- it does not raise exception in case of problems, only returns `false`.

Reply:
Done in b64cfed4e5a8b51f9546fc04b14bb9c81421306f